### PR TITLE
Show status of auto updating lists

### PR DESCRIPTION
**Pull request by ray-lee**
## Description

This modifies the UI so that users can see the status of auto updating lists.

- The entries list is now displayed next to the search results list, even when the list is auto updating. Previously, it was only shown for manually updated lists.
- A status and description are now shown at the top of the entries list, if the list is auto updating. This changes based on the `auto_update_status` value returned from the CM, as well as the local modification state of the search parameters in the list editor. 
   auto_list_status
- In order to implement a different behavior when search parameters have been modified, there is now an `isSearchModified` property in the Redux state for the list editor. Previously, there was only an `isModified` property, which tracked changes to any part of the list, possibly including the search parameters. The `isSearchModified` property now separately tracks only the changes to search parameters, and the `isModified` computation now uses the value of `isSearchModified` instead of checking for changes to search parameters itself.
- Because the CM does not handle changing a list from manual to auto after it has been saved, the auto/manual radio buttons are now enabled only when creating a new list. Once a list is saved, they are disabled.

The possible statuses are:

**When a new list is being created:**

> Status: New
> 

> *The system will begin to populate this list using the configured search criteria when it is saved, and will fully populate the list during the first scheduled update that occurs after it has been saved.*
> 

**When a saved list has `auto_update_state` = `init`:**

> Status: Initializing
> 

> *This list was created recently. The system has partially populated the list using the configured search criteria, and will fully populate the list during the next scheduled update.*
> 

**When a saved list has `auto_update_state` = `repopulate`:**

> Status: Repopulating
> 

> *The search criteria for this list were changed recently, but the entries have not yet been updated. The system will repopulate the list using the current search criteria during the next scheduled update.*
> 

**When a saved list has `auto_update_state` = `updated`:**

> Status: Updated
> 

> *This list was fully populated during the last scheduled update, using the configured search criteria and the titles that were available at the time. Titles that have been acquired since the last update will be added to the list during the next scheduled update.*
> 

**When a saved list has unsaved changes to its search parameters (regardless of  `auto_update_state`)**

> Status: Search criteria modified
> 

> *There are unsaved changes to the search criteria for this list. The system will repopulate the list using the new search criteria during the first scheduled update that occurs after it has been saved.*
>

## Motivation and Context

This makes it easier for users to understand how auto updating lists work, and why entries in a list may not match the current search results.

Notion: https://www.notion.so/lyrasis/Implement-Admin-UI-changes-for-auto-updating-lists-a65aeb9b85214073ace9818deb04a17e

## How Has This Been Tested?

1. Create a new auto-updating list. Verify that the list status is "New".
2. Save the list.  Verity that the list status changes to "Initializing", and that the auto/manual radio buttons are disabled.
3. Wait until the CM has updated the list, and re-open the list. Verify that the list status changes to "Updated".
4. Change the search filters. Verify that the list status changes to "Search criteria modified".
5. Save the list with the new search filters. Verify that the list status changes to "Repopulating".
6. Wait until the CM has updated the list, and re-open the list. Verify that the list status changes to "Updated".
7. Create a new manual list. Verify that the list status disappears when the Manual radio button is selected.
8. Open a saved manual list. Verify that the list status is not displayed.

## Checklist:

- [x] I have updated the documentation accordingly. https://www.notion.so/lyrasis/Admin-UI-Auto-Updating-Lists-935ccb82628a49ba940c7ee97c75039a
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by tdilauro — COMMENTED** [review]
This looks pretty good.

There is one assertion in the `updated` status text that I wanted to clarify, regarding which titles are eligible when a list update is run.

I also suggested some possible UI text rewording for consistency, but they are not blockers for me.

**Comment by tdilauro** [comment]
Might also be helpful to include the list of statuses in the main PR comment, since some with access to this repo will not have access to Notion.

File: src/components/CustomListEntriesEditor.tsx [hunk 1]
```diff
@@ -256,125 +260,163 @@ const CustomListEntriesEditor = ({
...
+      autoUpdateStatusDescription =
+        "This list was created recently. The system has partially populated the list using the configured search criteria, and will fully populate the list during the next scheduled update.";
+    } else if (autoUpdateStatus === "repopulate") {
+      autoUpdateStatusName = "Repopulating";
+      autoUpdateStatusDescription =
+        "The search criteria for this list were changed recently, but the entries have not yet been updated. The system will repopulate the list using the current search criteria during the next scheduled update.";
+    } else if (autoUpdateStatus === "updated") {
+      autoUpdateStatusName = "Updated";
+      autoUpdateStatusDescription =
+        "This list was fully populated during the last scheduled update, using the configured search criteria and the titles that were available at the time. Titles that have been acquired since the last update will be added to the list during the next scheduled update.";
```
[the remark this answers is not in the document]
Comment (ray-lee) [reply src/components/CustomListEntriesEditor.tsx:296]: I think it depends on which update. If it's the first one, then all titles are considered. After that, only the new titles are considered. In either case, it should result in the list being updated to reflect all the titles that are available at the time. I was trying to describe that without getting too far into the details, but I'm open to rewordings.

**Review by tdilauro — APPROVED** [review]
Looks good! 🚀

**Pull request merged**